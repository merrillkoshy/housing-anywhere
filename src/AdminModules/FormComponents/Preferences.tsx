import React, { useState } from "react";
import { Form, Dropdown } from "react-bootstrap";

const Preferences = () => {
	const [minAge, setMinAge] = useState(0);
	const [maxAge, setMaxAge] = useState(0);
	const [preferredGender, setPreferredGender] = useState("No Preference");

	const [alias, setAlias] = useState("");
	const [externalReference, setExternalReference] = useState("");
	const [extraData, setExtraData] = useState("");

	return (
		<>
			<Form.Group className="mb-3">
				<Form.Label>Min Age</Form.Label>
				<Form.Control
					type="number"
					placeholder="Enter minAge"
					value={minAge}
					id="minAge"
					min={18}
					onChange={(e) => {
						e.preventDefault();
						setMinAge((e.currentTarget as HTMLInputElement).valueAsNumber);
					}}
				/>
			</Form.Group>
			<Form.Group className="mb-3">
				<Form.Label>Max Age</Form.Label>
				<Form.Control
					type="number"
					placeholder="Enter maxAge"
					value={maxAge}
					min={35}
					id="maxAge"
					onChange={(e) => {
						e.preventDefault();
						setMaxAge((e.currentTarget as HTMLInputElement).valueAsNumber);
					}}
				/>
			</Form.Group>
			<Form.Group className="mb-3">
				<Dropdown>
					<Dropdown.Toggle variant="primary" id="preferredGender">
						{preferredGender}
					</Dropdown.Toggle>

					<Dropdown.Menu>
						<Dropdown.Item
							onClick={(e) => {
								e.preventDefault();
								setPreferredGender("male");
							}}
						>
							Male
						</Dropdown.Item>
						<Dropdown.Item
							onClick={(e) => {
								e.preventDefault();
								setPreferredGender("female");
							}}
						>
							Female
						</Dropdown.Item>
						<Dropdown.Item
							onClick={(e) => {
								e.preventDefault();
								setPreferredGender("other");
							}}
						>
							Other
						</Dropdown.Item>
					</Dropdown.Menu>
				</Dropdown>
			</Form.Group>
			<Form.Group className="mb-3">
				<Form.Label>Alias</Form.Label>
				<Form.Control
					type="text"
					id="alias"
					value={alias}
					onChange={(e) => {
						e.preventDefault();
						setAlias(e.currentTarget.value);
					}}
					placeholder="Enter Alias"
				/>
			</Form.Group>
			<Form.Group className="mb-3">
				<Form.Label>External Reference</Form.Label>
				<Form.Control
					type="text"
					id="externalReference"
					value={externalReference}
					onChange={(e) => {
						e.preventDefault();
						setExternalReference(e.currentTarget.value);
					}}
					placeholder="Enter External Reference"
				/>
			</Form.Group>
			<Form.Group className="mb-3">
				<Form.Label>Extra Data</Form.Label>
				<Form.Control
					type="text"
					id="extraData"
					value={extraData}
					onChange={(e) => {
						e.preventDefault();
						setExtraData(e.currentTarget.value);
					}}
					placeholder="A JSON field with extra information important for the listing which doesn't fit in other fields. Make sure to escape the quotes!"
				/>
			</Form.Group>
		</>
	);
};
export default Preferences;
