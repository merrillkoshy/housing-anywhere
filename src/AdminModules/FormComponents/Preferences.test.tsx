import React from "react";
import { shallow } from "enzyme";
import Preferences from "./Preferences";

describe("Preferences module", () => {
	const PreferencesModule = shallow(<Preferences />);
	it("renders", () => {
		expect(PreferencesModule).toMatchSnapshot();
	});
	it("finds a minAge input which is a number", () => {
		expect(PreferencesModule.find("#minAge").exists()).toBe(true);
	});
	it("finds a maxAge input which is a number", () => {
		expect(PreferencesModule.find("#maxAge").exists()).toBe(true);
	});
	it("finds a preferredGender input which is a string", () => {
		expect(PreferencesModule.find("#preferredGender").exists()).toBe(true);
	});
	it("finds a alias input which is a string", () => {
		expect(PreferencesModule.find("#alias").exists()).toBe(true);
	});
	it("finds a externalReference input which is a string", () => {
		expect(PreferencesModule.find("#externalReference").exists()).toBe(true);
	});
	it("finds a extraData input which is a string", () => {
		expect(PreferencesModule.find("#extraData").exists()).toBe(true);
	});
});
