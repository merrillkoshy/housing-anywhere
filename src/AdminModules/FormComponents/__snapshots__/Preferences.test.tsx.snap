// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Preferences module renders 1`] = `
<Fragment>
  <FormGroup
    className="mb-3"
  >
    <FormLabel
      column={false}
      srOnly={false}
    >
      Min Age
    </FormLabel>
    <FormControl
      id="minAge"
      min={18}
      onChange={[Function]}
      placeholder="Enter minAge"
      type="number"
    />
  </FormGroup>
  <FormGroup
    className="mb-3"
  >
    <FormLabel
      column={false}
      srOnly={false}
    >
      Max Age
    </FormLabel>
    <FormControl
      id="maxAge"
      min={35}
      onChange={[Function]}
      placeholder="Enter maxAge"
      type="number"
    />
  </FormGroup>
  <FormGroup
    className="mb-3"
  >
    <Dropdown
      navbar={false}
    >
      <DropdownToggle
        id="preferredGender"
        variant="primary"
      />
      <DropdownMenu
        align="left"
        alignRight={false}
        flip={true}
      >
        <DropdownItem
          as={
            Object {
              "$$typeof": Symbol(react.forward_ref),
              "render": [Function],
            }
          }
          disabled={false}
          onClick={[Function]}
        >
          Male
        </DropdownItem>
        <DropdownItem
          as={
            Object {
              "$$typeof": Symbol(react.forward_ref),
              "render": [Function],
            }
          }
          disabled={false}
          onClick={[Function]}
        >
          Female
        </DropdownItem>
        <DropdownItem
          as={
            Object {
              "$$typeof": Symbol(react.forward_ref),
              "render": [Function],
            }
          }
          disabled={false}
          onClick={[Function]}
        >
          Other
        </DropdownItem>
      </DropdownMenu>
    </Dropdown>
  </FormGroup>
  <FormGroup
    className="mb-3"
  >
    <FormLabel
      column={false}
      srOnly={false}
    >
      Alias
    </FormLabel>
    <FormControl
      id="alias"
      onChange={[Function]}
      placeholder="Enter Alias"
      type="text"
    />
  </FormGroup>
  <FormGroup
    className="mb-3"
  >
    <FormLabel
      column={false}
      srOnly={false}
    >
      External Reference
    </FormLabel>
    <FormControl
      id="externalReference"
      onChange={[Function]}
      placeholder="Enter External Reference"
      type="text"
    />
  </FormGroup>
  <FormGroup
    className="mb-3"
  >
    <FormLabel
      column={false}
      srOnly={false}
    >
      Extra Data
    </FormLabel>
    <FormControl
      id="extraData"
      onChange={[Function]}
      placeholder="A JSON field with extra information important for the listing which doesn't fit in other fields. Make sure to escape the quotes!"
      type="text"
    />
  </FormGroup>
</Fragment>
`;
